src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Log/Logger.cs

```diff
@@ -57,7 +55,7 @@ public static void Log(FunctionId functionId, string message = null)
                 return;
             }
 
-            logger.Log(functionId, LogMessage.Create(message));
+            logger.Log(functionId, LogMessage.Create(message ?? ""));
```

Comment:
📝 Passing `null` here ends up on a path where the `StaticLogMessage` never gets put back in the object pool.